### PR TITLE
Auto-Hardware Wallet Connections Removed

### LINKED ISSUE #17605: [Bug]: Automatic Hardware Wallet Connection
### Describe the bug

Metamask (MV2) will optimistically (forcefully) connect to a previously connected hardware wallet. Meaning that there can be some difficulties for those who own multiple hardware wallets, e.g. a Trezor and a Ledger, and want both of them connected on MM.

### Steps to reproduce

1. Import a Ledger hardware wallet account 
image
2. Add 1-2 ledger accounts on the account select page
3. Attempt to connect to a Trezor device and you'll be forcefully taken to the Ledger account selection list

### Error messages or log output

_No response_

### Version

10.24.2

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

Ledger, Trezor

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by andrewpeters9**
## Explanation

* Fixes #17605 

## Manual Testing Steps

- Connect a Ledger account
- Revisit the Connect Hardware wallet page
- If the page doesn't automatically connect you to your Ledger then the fix has worked

- To carry out further QA: 
  - Play around with connecting to Trezors/Lattices/Ledgers
  - Experiment with importing accounts, cancelling connections, connecting to locked devices etc

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue

### CODE REVIEW

File: ui/pages/create-account/connect-hardware/index.js [hunk 1]
```diff
@@ -72,44 +67,16 @@ class ConnectHardwareForm extends Component {
...
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const { accounts } = nextProps;
-    const newAccounts = this.state.accounts.map((a) => {
-      const normalizedAddress = a.address.toLowerCase();
-      const balanceValue = accounts[normalizedAddress]?.balance || null;
-      a.balance = balanceValue ? formatBalance(balanceValue, 6) : '...';
-      return a;
-    });
-    this.setState({ accounts: newAccounts });
-  }
```
Comment (andrewpeters9) [inline ui/pages/create-account/connect-hardware/index.js:84]: Unsafe method removed as all it was doing is creating a `computed` property. Method is easily replaced with the line in `account-list.js`:

```js
accounts: formatAccounts(props.accounts, getMetaMaskAccounts(state)),
```

File: ui/pages/create-account/connect-hardware/index.js [hunk 2]
```diff
@@ -72,44 +67,16 @@ class ConnectHardwareForm extends Component {
...
-
-  componentDidMount() {
-    this.checkIfUnlocked();
```
Comment (andrewpeters9) [inline ui/pages/create-account/connect-hardware/index.js:87]: Removed as we don't need to check if all devices are unlocked upon mount.

File: ui/pages/create-account/connect-hardware/index.js [hunk 3]
```diff
@@ -72,44 +67,16 @@ class ConnectHardwareForm extends Component {
...
-  }
-
-  async checkIfUnlocked() {
-    for (const device of [
-      HardwareDeviceNames.trezor,
-      HardwareDeviceNames.ledger,
-      HardwareDeviceNames.lattice,
-    ]) {
```
Comment (andrewpeters9) [inline ui/pages/create-account/connect-hardware/index.js:95]: Removed as we don't have a scenario where we need to check that all devices are unlocked and then subsequently fetch the first page of their accounts.

File: ui/pages/create-account/connect-hardware/index.js [hunk 4]
```diff
@@ -156,19 +123,8 @@ class ConnectHardwareForm extends Component {
...
-          // Map accounts with balances
-          const newAccounts = accounts.map((account) => {
-            const normalizedAddress = account.address.toLowerCase();
-            const balanceValue =
-              this.props.accounts[normalizedAddress]?.balance || null;
-            account.balance = balanceValue
-              ? formatBalance(balanceValue, 6)
-              : '...';
-            return account;
-          });
```
Comment (andrewpeters9) [inline ui/pages/create-account/connect-hardware/index.js:168]: Moved to `util.ts`

File: ui/pages/create-account/connect-hardware/index.js [hunk 5]
```diff
@@ -308,27 +264,28 @@ class ConnectHardwareForm extends Component {
...
-        device={this.state.device}
-        accounts={this.state.accounts}
-        connectedAccounts={this.props.connectedAccounts}
-        selectedAccounts={this.state.selectedAccounts}
+        device={device}
+        accounts={accounts}
+        selectedAccounts={selectedAccounts}
         onAccountChange={this.onAccountChange}
-        chainId={this.props.chainId}
-        rpcPrefs={this.props.rpcPrefs}
```
Comment (andrewpeters9) [inline ui/pages/create-account/connect-hardware/index.js:331]: Some props here are removed as they can be easily added via a separate `redux` connection

**Comment by andrewpeters9** [comment]
Marked as draft until e2e tests pass

**Comment by CharlesBlaqk** [comment]
We'd be happy to look into this for you. kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

**Pull request closed without merging**